File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/005-ADDN-Addon Node.swift
//
//  ADDNRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class ADDNRecord: Record {
    public var description: String { return "ADDN: \(EDID)" }
    public var EDID: STRVField // Editor ID
    public var CNAME: CREFField // RGB color

    override func createField(r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "EDID": EDID = STRVField(r, dataSize)
        case "CNAME": CNAME = CREFField(r, dataSize)
        default: return false
        }
        return true
    }
}

File: src/ObjectManager/Object.Tes/FilePacks/Records/005-ADDN-Addon Node.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class ADDNRecord : Record
    {
        public override string ToString() => $"ADDN: {EDID.Value}";
        public STRVField EDID { get; set; } // Editor ID
        public CREFField CNAME; // RGB color

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            switch (type)
            {
                case "EDID": EDID = r.ReadSTRV(dataSize); return true;
                case "CNAME": CNAME = r.ReadT<CREFField>(dataSize); return true;
                default: return false;
            }
        }
    }
}